p/scripts/user.query.js

```diff
@@ -13,32 +13,73 @@ function init_draggable() {
 	
 	let source;
 	const configureQueries = document.querySelector('#configureQueries');
+	const addMarker = (position, element) => {
+		const hr = configureQueries.querySelector('hr.drag-drop-marker');
+		if (null === hr) {
+			element.insertAdjacentHTML(position, '<hr class="drag-drop-marker" />');
+		}
+	};
+	const removeMarker = () => {
+		const hr = configureQueries.querySelector('hr.drag-drop-marker');
+		if (null !== hr) {
+			hr.remove();
+		}
+	};
 
 	configureQueries.addEventListener('dragstart', event => {
 		source = event.target.closest('[draggable="true"]');
 		event.dataTransfer.setData('text/html', source.outerHTML);
 		event.dataTransfer.effectAllowed = 'move';
 	});
-	configureQueries.addEventListener('dragover', event => event.preventDefault());
-	configureQueries.addEventListener('dragleave', event => event.preventDefault());
+	configureQueries.addEventListener('dragover', event => {
+		event.preventDefault();
+		if ('function' !== typeof event.target.closest) {
+			return;
+		}
+
+		const dropQuery = event.target.closest('[draggable="true"]');
+		if (null === dropQuery || source === dropQuery) {
+			return;
+		}
+
+		const rect = dropQuery.getBoundingClientRect();
+		if (event.clientY < (rect.top + rect.height / 2)) {
+			addMarker('beforebegin', dropQuery);
+		} else {
+			addMarker('afterend', dropQuery);
+		}
+	});
+	configureQueries.addEventListener('dragleave', event => {
+		event.preventDefault();
+		removeMarker();
+	});
 	configureQueries.addEventListener('drop', event => {
 		event.preventDefault();
 		event.stopPropagation();
+		if ('function' !== typeof event.target.closest) {
+			return;
+		}
+
 		const dropQuery = event.target.closest('[draggable="true"]');
-		if (null === dropQuery) {
-			source.remove();
-			configureQueries.querySelector('legend').insertAdjacentHTML('afterend', event.dataTransfer.getData('text/html'));
-		} else if (source !== dropQuery) {
-			source.remove();
+		if (null === dropQuery || source === dropQuery) {
+			return;
+		}
+
+		const rect = dropQuery.getBoundingClientRect();
+		if (event.clientY < (rect.top + rect.height / 2)) {
+			dropQuery.insertAdjacentHTML('beforebegin', event.dataTransfer.getData('text/html'));
+		} else {
 			dropQuery.insertAdjacentHTML('afterend', event.dataTransfer.getData('text/html'));
 		}
+		source.remove();
+		removeMarker();
 	});
 
 	// This is needed to work around a Firefox bug → https://bugzilla.mozilla.org/show_bug.cgi?id=800050
 	configureQueries.addEventListener('focusin', event => {
 		event.target.closest('input[id^="queries_"][id$="_name"]').select();
```

Comment:
I can generate the following error (Firefox 84 / Ubuntu 20.10)
![image](https://user-images.githubusercontent.com/1008324/104364354-c6ed1600-5516-11eb-8464-2f10a0a2d057.png)

Reply:
When hitting the validate button

Reply:
Even though I have no idea how you get those, I add a fix.

Reply:
My fix is not correct

Reply:
This time, it should be working as intended.